Tools/SandboxTest.ps1

```diff
@@ -273,6 +305,132 @@ function Test-FileChecksum {
+        $cachedTokenIsEmpty = [string]::IsNullOrWhiteSpace($cachedTokenContent)
+
+        # It is possible for a token to be both empty and expired. Since these are debug and verbose messages, showing both doesn't hurt
+        if ($cachedTokenIsEmpty) { Write-Verbose 'Cached token had no content. It will be re-validated' }
+        if ($cacheIsExpired) { Write-Verbose "Cached token is older than $script:CachedTokenExpiration days. It will be re-validated" }
+
+        if (!$cacheIsExpired -and !$cachedTokenIsEmpty) {
+            # Check the content of the cached file in case the actual token expiration is known
+            Write-Verbose 'Attempting to fetch token expiration from cache'
+            # Since Github adds ` UTC` at the end, it needs to be stripped off. Trim is safe here since the last character should always be a digit
+            $cachedExpirationForParsing = $cachedTokenContent.TrimEnd(' UTC')
+            $cachedExpirationDate = [System.DateTime]::MinValue
+            [System.DateTime]::TryParse($cachedExpirationForParsing, [ref]$cachedExpirationDate)
+
+            $tokenExpirationDays = $cachedExpirationDate - (Get-Date) | Select-Object -ExpandProperty TotalDays
+
+            if ($cachedExpirationForParsing -eq [System.DateTime]::MaxValue.ToLongDateString().Trim()) {
+                Write-Verbose "The cached token contained content. It is set to never expire"
+                return $true
+            }
+
+            if ($tokenExpirationDays -gt 0) {
+                Write-Verbose "The cached token contained content. It should expire in $tokenExpirationDays days"
+                return $true
+            }
+            # If the parsing failed, the expiration should still be at the minimum value
+            elseif ($cachedExpirationDate -eq [System.DateTime]::MinValue) {
+                Write-Verbose 'The cached token contained content, but it could not be parsed as a date. It will be re-validated'
+                Invoke-FileCleanup -FilePaths $cachedToken.FullName
+                # Do not return anything, since the token will need to be re-validated
+            }
+            else {
+                Write-Verbose "The cached token contained content, but the token expired $([Math]::Abs($tokenExpirationDays)) days ago"
+                # Leave the cached token so that it doesn't throw script exceptions in the future
+                # Invoke-FileCleanup -FilePaths $cachedToken.FullName
+                return $false
+            }
+        }
+        else {
+            # Either the token was empty, or the cached token is expired. Remove the cached token so that re-validation
+            # of the token will update the date the token was cached if it is still valid
+            Invoke-FileCleanup -FilePaths $cachedToken.FullName
+        }
+    }
+    else {
+        Write-Verbose 'Token was not found in the cache'
```

Comment:
It's possible that a hash-file exists in the directory, but the user is using a different / new token. It may make sense to clean up all other files in the directory in this condition

Reply:
Some users (like me) may have multiple tokens. For example, I have an account through GitHub EMU and my personal account (this one). It's also possible for a sinle user to have multiple tokens with different permissions.

Or, its possible that some users may share a single PC and a single Windows login, but use different GitHub tokens.

This implementation specifically allows for multiple token hashes to be in the cache to account for these scenarios